enhancements/oc/icsp-registry-scoped.md

```diff
@@ -0,0 +1,245 @@
+will result in a new entry in their ICSP, and thus require a cluster reboot. It
+is also possible that any time the customer mirrors updates for the operators
+they already use, if any one of those updates references a container image from
+a new location (this applies to any image related to the operator or its
+operand), that also may result in a change to the ICSP and thus a reboot.
+
+Some clusters may take a long time to reboot, and some customers may be averse
+to rebooting more often than is necessary. Bare metal clusters in particular
+may take up to 15 minutes to reboot each Node, not including the time it takes
+to drain workloads. This enhancement enables a customer to mirror operators
+into their local registry as often as they want without inducing a reboot, so
+long as they are mirroring content from a registry that they have mirrored from
+in the past.
+
+### Goals
+
+* Enable a customer to add an operator to their local mirror without causing their cluster to reboot.
+* Enable a customer to mirror updates to operators they already use without causing their cluster to reboot.
+
+### Non-Goals
+
+* Eliminate the need to reboot a cluster when an ImageContentSourcePolicy changes or is created.
+
+## Proposal
+
+Change the `oc adm mirror` command so that the ImageContentSourcePolicy it
+outputs is scoped to the registry level. It will not include any path
+information.
+
+Optionally a command-line flag could be used to enable today's behavior of
+scoping ICSP entries to a specific repository, in case that behavior is
+desirable to someone.
+
+### User Stories
+
+#### Story 1
+
+As a user with a cluster in a disconnected network, I can mirror optional
+operators into my local registry, and my cluster will only need to reboot Nodes
+the first time I mirror content from any particular remote registry.
+
+#### Story 2
+
+As a user with a cluster in a disconnected network, I can mirror updates to
+optional operators that I already have installed without causing my cluster to
+reboot.
+
+### Implementation Details/Notes/Constraints [optional]
+
+This is an example of a registry-scoped ICSP:
+
+```
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: redhat-operators
+spec:
+  repositoryDigestMirrors:
+  - mirrors:
+    - local.registry:5000
+    source: registry.redhat.io
+```
+
+The above ICSP was tested on OCP 4.4, and it resulted in an entry as shown
+below in
+[`/etc/containers/registries.conf`](https://github.com/containers/image/blob/8051f86/docs/containers-registries.conf.5.md)
+on each Node:
+
+```
+[[registry]]
+  prefix = ""
+  location = "registry.redhat.io"
+  mirror-by-digest-only = true
+
+  [[registry.mirror]]
+    location = "local.registry:5000"
+```
+
+An operator was then installed, and its image was observed being pulled from
+`local.registry:5000`.
+
+### Risks and Mitigations
+
+No new risks are introduced. This enhancement proposes to use an existing API
```

Comment:
It does seem like there is some small risk - we may introduce an overly broad ICSP that applies to images that users didn't intend or expect, and the mitigation would be the option to produce the more specific ICSP.